logging/logging.go

```diff
@@ -43,3 +46,21 @@ func (f *textFormatter) Format(entry *log.Entry) ([]byte, error) {
 	b.WriteByte('\n')
 	return b.Bytes(), nil
 }
+
+// WithContext returns a logger that has information from the context as fields.
+//
+// e.g.
+//     logger := logging.WithContext(ctx)
+//     logger.Errorf("Some error")
+func WithContext(ctx context.Context) *log.Entry {
```

Comment:
I figure if ever we add more information to the context, we might want to add more information here. `WithStandardWeaveworksContextInfo` is a bit wordy.

I guess instead we could add more `WithFoo` functions, since they do chain.